### PR TITLE
shell script changes for linux 7

### PR DESCRIPTION
**Pull request by manisha1331-zz**
Changes for installing Jwala on Oracle Linux 7.
Removing Oracle Linux 6 support.

### CODE REVIEW

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 1]
```diff
@@ -12,115 +12,27 @@
 ### END INIT INFO
 
 # Source LSB function library.
-. /etc/init.d/functions
 
-NAME=`basename "$0"`
+[Unit]
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh:1]: Delete all the lines above this one

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 2]
```diff
@@ -12,115 +12,27 @@
...
+Description= jwala service file
+After=network.target
 
-TOMCAT_HOME=@TOMCAT_HOME@
-TOMCAT_HOME_BIN=$TOMCAT_HOME/bin
+[Service]
+Type=forking
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh:6]: Add a PIDFile option as well when using forking.
The CATALINA_PID is already being set so just use the same file.

See the 'forking' description here:
https://www.freedesktop.org/software/systemd/man/systemd.service.html

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 3]
```diff
@@ -12,115 +12,27 @@
...
 
 # Define the tomcat username
-TOMCAT_USER="${TOMCAT_USER:-tomcat}"
+User=tomcat
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh]: Add the Group option as well, and the value should be a parameter (TOMCAT_GROUP)

Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh]: tomcat should be a parameter (TOMCAT_USER) that gets replaced like TOMCAT_HOME.
Check install-service.sh and the previous implementation of this script to see how it was done.

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 4]
```diff
@@ -12,115 +12,27 @@
...
 
-# Define the tomcat group
-TOMCAT_GROUP="${TOMCAT_GROUP:-`id -gn $TOMCAT_USER`}"
+#Environment="JAVA_HOME=/usr/lib/jvm/jre"
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh]: Remove any commented out lines

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 5]
```diff
@@ -12,115 +12,27 @@
...
 
-CATALINA_PID=$TOMCAT_HOME/logs/catalina.pid
-export CATALINA_PID
+Environment="CATALINA_BASE=@TOMCAT_HOME@"
+Environment="CATALINA_HOME=@TOMCAT_HOME@"
+Environment="CATALINA_PID=@TOMCAT_HOME@/logs/catalina.pid"
+Environment="CATALINA_HOME_BIN=@TOMCAT_HOME@/bin"
 
-RETVAL="0"
+ExecStart=/opt/ctp/jwala-0.0.245/apache-tomcat-7.0.55/bin/startup.sh
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh]: These paths can't be hardcoded.
See the install-service.sh for how tokens were replaced in the previous version of the script.

File: jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh [hunk 6]
```diff
@@ -12,115 +12,27 @@
...
+ExecStop=/opt/ctp/jwala-0.0.245/apache-tomcat-7.0.55/bin/catalina.sh
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/linux/jvm-service.sh]: Add the stop parameter at the end of this line, or use the shutdown.sh script

File: jwala-tomcat/src/main/resources/data/scripts/install-service.sh [hunk 7]
```diff
@@ -47,10 +51,23 @@ if $cygwin; then
...
+  get_version=$(uname -r)
+  linux_7="el7"
+  service_file=$1".service"
+  if [[ $get_version =~ $linux_7 ]];then
+    pushd $(dirname $0)
+    chown -R ${TOMCAT_USER}:${TOMCAT_GROUP} $TOMCAT_HOME/work
+    chown -R ${TOMCAT_USER}:${TOMCAT_GROUP} $TOMCAT_HOME/logs
+    chown -R ${TOMCAT_USER}:${TOMCAT_GROUP} $TOMCAT_HOME/temp
+    chown -R ${TOMCAT_USER}:${TOMCAT_GROUP} $TOMCAT_HOME/data
+    /bin/sed -e "s/@TOMCAT_HOME@/${2//\//\\/}\\/$1\\/$3/g" -e "s/@JVM_NAME@/$1/g" linux/jvm-service.sh> $service_file
```
Comment (jmahmoodcc) [inline jwala-tomcat/src/main/resources/data/scripts/install-service.sh]: Here is where the token replacement happens for the service unit template

**Review by jmahmoodcc — DISMISSED** [review]
I don't have access to push changes to your fork. 
Can you please add me as a collaborator? I have a bunch of minor changes in your branch that I want to push.

Focus on the linux/jvm-service.sh and install-service.sh changes that are needed.

**Comment by dskcode** [comment]
> I don't have access to push changes to your fork.
> Can you please add me as a collaborator? I have a bunch of minor changes in your branch that I want to push.
> 
> Focus on the linux/jvm-service.sh and install-service.sh changes that are needed.


@jmahmoodcc, @manisha1331 did add you as a collaborator to her fork on Friday. You must have received an invite over email, you just need to accept and you should be good to go...

**Comment by jmahmoodcc** [comment]
I accepted the invite when she sent it, before I made the changes.
I'll continue looking into it from my end.

**Comment by jmahmoodcc** [comment]
Figured it out.
Looks like HTTPS authentication wasn't being accepted but setting up SSH keys worked.

**Comment by jmahmoodcc** [comment]
@dskcode @Karthik263 
Please review as soon as possible. Thanks!

**Pull request merged**